### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by itepifanio**
This MR updates the README.md and API docs.

- Improved the statement of need on the README.md
- Added follow-up instructions after the installation section
- Remove hard-coded dependencies from README.md
- Enhancements were made in the API docs section, explaining better the imports and all modules related.
- Added section explaining how to run the tests

@ibayer I'll open a second PR to address the [JOSS feedback](https://github.com/openjournals/joss-reviews/issues/4480#issuecomment-1179173615) on the paper.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,13 @@
 
 ![CI-Badge](https://github.com/palaimon/ipyannotator/workflows/CI/badge.svg)
 
-Ipyannotator is a flexible annotation system. The library contains some pre-defined annotators that can be used out of the box, but it also can be extend and customized according to the users needs.
+Ipyannotator is a flexible annotation system. Developed to allow users to hack its features by extending and customizing it. 
+
+The large variety of annotation tasks, data formats and data visualizations is a challenging when dealing with multiple domains of supervised machine learning (ML). The existent tooling is not flexible enough which imposes limitations to the user. By providing a framework where users can use, customize and create their own annotation tooling this projects aims to solve this problem.
```
Comment (ibayer) [inline README.md:7]: > The existent tooling is not flexible

The existent tooling is **often** not flexible

File: README.md [hunk 2]
```diff
@@ -14,59 +20,50 @@ Please let us know if you find this repository useful. Your feedback will help u
...
-ipyevents = "^0.8.0"
-ipywidgets = "^7.5.1"
-```
+Ipyannotator provides a [simple API](https://palaimon.github.io/ipyannotator/nbs/21_api_doc.html) that provides the ability to explore, create and improve annotation datasets by using a pair of input/outputs. All pair of input/output are [listed on Ipyannotator's docs](https://palaimon.github.io/ipyannotator/nbs/22_input_output_doc.html). Check [Ipyannotator tutorials](https://palaimon.github.io/ipyannotator/docs/tutorials.html) for a quickly demonstration of the library.
 
-## Run ipyannotator as stand-alone web app using voila
+### Run ipyannotator tests
 
-Using `poetry`:
+The notebooks contains the development dependencies: [pytest](https://docs.pytest.org/en/7.1.x/) and [ipytest](https://github.com/chmp/ipytest). The tests can be executed using `nbdev_test_nbs` on the terminal.
```
Comment (ibayer) [inline README.md:33]: > The tests can be executed using `nbdev_test_nbs` on the terminal.

Again put this first. Also, if dev requirements are needed to run the tests please doc this here.

File: README.md [hunk 3]
```diff
@@ -14,59 +20,50 @@ Please let us know if you find this repository useful. Your feedback will help u
...
 
-## Run ipyannotator as stand-alone web app using voila
+### Run ipyannotator tests
 
-Using `poetry`:
+The notebooks contains the development dependencies: [pytest](https://docs.pytest.org/en/7.1.x/) and [ipytest](https://github.com/chmp/ipytest). The tests can be executed using `nbdev_test_nbs` on the terminal. 
+
+### Run ipyannotator as stand-alone web app using voila
+
+Ipyannotator can be executed as a web app using the [voila](https://github.com/voila-dashboards/voila) library. The following sections describe how to run using poetry and pip.
```
Comment (ibayer) [inline README.md:37]: > describe how to run using poetry and pip.

describe how to run the voila app using poetry or pip.

**Review by ibayer — CHANGES_REQUESTED** [review]
@itepifanio LGTM I have just added a few suggestions for minor edits.

File: README.md (hunk 3 above)
Comment (itepifanio) [reply README.md:39]: @ibayer the following sections describe how to run using poetry or pip.

Let me know if your previous suggestions were also solved.

**Comment by ibayer** [comment]
@itepifanio thanks

**Pull request merged**